### PR TITLE
DIMAP: Parse and emit geometric metadatas (Fixes #6343)

### LINKED ISSUE #6343: DIMAP based drivers miss some useful geometric values
## Expected behavior and actual behavior.

I believe that the Spot and Pleiades metadata drivers miss some useful properties (like sun and viewing angles).
`gdalinfo` should be able to display these MDs, and a DIMAP product converted into, say a geotiff, should expose these MDs.

## Steps to reproduce the problem.

`gdalinfo /path/to/dimap/PROD_.../VOL_.../IMG_.../DIM_SPOT6_..._1.XML` does not show the following metadata properties:
| Name | Path |
| ---- | ---- |
| Azimuth angle | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.AZIMUTH_ANGLE |
| Viewing angle | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.VIEWING_ANGLE |
| Viewing angle across track | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.VIEWING_ANGLE_ACROSS_TRACK |
| Viewing angle along track | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.VIEWING_ANGLE_ALONG_TRACK |
| Indicence angle | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.INCIDENCE_ANGLE |
| Indicence angle across track | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.INCIDENCE_ANGLE_ACROSS_TRACK |
| Indicence angle along track | Geometric_Data.Use_Area.Located_Geometric_Values.Acquisition_Angles.INCIDENCE_ANGLE_ALONG_TRACK |
| Sun azimuth | Geometric_Data.Use_Area.Located_Geometric_Values.Solar_Incidences.SUN_AZIMUTH |
| Sun elevation | Geometric_Data.Use_Area.Located_Geometric_Values.Solar_Incidences.SUN_ELEVATION |

There are several occurrences of the geometric data, but maybe adding the values for the central ground point could be useful for many users/developers?

## Operating system

Ubuntu 20.04

## GDAL version and provenance

GDAL 3.4.2, released 2022/03/08 compiled from sources

### PR DESCRIPTION
**Pull request by tbonfort**
## What does this PR do?

Fixes #6343

### CODE REVIEW

File: frmts/dimap/dimapdataset.cpp [hunk 1]
```diff
@@ -1550,6 +1551,33 @@ int DIMAPDataset::ReadImageInformation2()
...
+    CPLXMLNode *psLocatedUseAreaNode =
+        CPLGetXMLNode(
+            psDoc,
+            "Geometric_Data.Use_Area");
+    if(psLocatedUseAreaNode!=nullptr) {
+        CPLXMLNode *psLocatedGeometricValuesNode = psLocatedUseAreaNode->psChild;
+        while (psLocatedGeometricValuesNode != nullptr)
+        {
+            CPLXMLNode *psLocationType = CPLGetXMLNode(psLocatedGeometricValuesNode, "LOCATION_TYPE");
+            if (psLocationType==nullptr || !EQUAL(psLocationType->psChild->pszValue, "center"))
```
Comment (rouault) [inline frmts/dimap/dimapdataset.cpp:1563]: for robustness against hostile datasets:
```suggestion
            if (psLocationType==nullptr || psLocationType->psChild == nullptr || !EQUAL(psLocationType->psChild->pszValue, "center"))
```

**Comment by tbonfort** [comment]
Thanks @rouault . Could this be a 3.5 backport candidate ?

**Comment by tbonfort** [comment]
Ok thanks. Are you planning for a 3.5.3 or is 3.6 the next expected release?

**Comment by rouault** [comment]
there will be a last 3.5.3 point release just one week before 3.6.0 . I tend to be a bit more conservative of what goes into the last point release in a series.

**Comment by remicres** [comment]
Thanks a lot!
I just have tried with the changes in `dimapdataset.cpp` and I works as expected.

**Pull request merged**